lib/spack/docs/environments.rst

```diff
@@ -433,6 +433,18 @@ The supplied location will become the build-directory for that package in all fu
     developers to only redirect the build directory if they understand their package's
     build-system.
 
+
+Spack attempts to minimize rebuilds by checking the source directory for the last time a file
+was changed and comparing it to the installation time.
+On some filesystems, such as lustre or a NFS, this check can be quite expensive.
+Users can set the configuration option ``config:dev_specs_always_rebuild:true`` to
+disable this check.
+``config:dev_specs_always_rebuild:true`` makes it so all develop specs will perform 
+**incremental rebuilds** each time ``spack install`` is called.  Please note these are not *complete rebuilds* and will still be much quicker than a fresh installation.
```

Comment:
These builds are only incremental for build systems that support incremental builds, autotools packages will have to do the entire build over again.

Reply:
The more you know... :+1: